test/smartwallet/customSmartWallet.test.ts

```diff
@@ -733,3 +733,564 @@
 // );
 
 // /* TODO - ADD tests for recover method */
```

Comment:
@AndresQuijano I left the whole file as it was before to let you check if additional tests have been added. Could you please review it and eventually add the missing cases and afterward delete the commented part?